dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/orm/EntityParser.java

```diff
@@ -0,0 +1,747 @@
+
+import io.openliberty.data.internal.persistence.Util;
+import io.openliberty.data.internal.persistence.orm.Models.AccessType;
+import io.openliberty.data.internal.persistence.orm.Models.Attribute;
+import io.openliberty.data.internal.persistence.orm.Models.AttributeKind;
+import io.openliberty.data.internal.persistence.orm.Models.Converter;
+import io.openliberty.data.internal.persistence.orm.Models.EmbeddableRecord;
+import io.openliberty.data.internal.persistence.orm.Models.EntityRecord;
+import io.openliberty.data.internal.persistence.orm.Models.MappedSuperclass;
+import jakarta.data.exceptions.MappingException;
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Convert;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Embedded;
+import jakarta.persistence.EmbeddedId;
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+
+/**
+ * Parses entity(ies) and creates an Object-Relational Model in memory
+ * that includes managed objects that do not have the Jakarta Persistence
+ * annotations required to be automatically modeled by the underlying Jakarta
+ * Persistence provider.
+ *
+ * This is an add-on feature of the Liberty Jakarta Data provider to support
+ * record entities which cannot be annotated, and for users who write Repository
+ * classes for java objects that comes from third party libraries which cannot
+ * be annotated.
+ */
+public class EntityParser {
+    private static final TraceComponent tc = Tr.register(EntityParser.class);
+
+    // ORM sets
+    private final Map<Class<?>, MappedSuperclass> mappedSuperclasses;
+    private final Map<Class<?>, EntityRecord> entities;
+    private final Map<Class<?>, EmbeddableRecord> embeddables;
+    private final Map<Class<?>, Converter> converters;
+
+    // Relationship tracking
+    private final Map<Class<?>, Class<?>> entityToRecord;
+    private final ConcurrentMap<Class<?>, Set<Class<?>>> entitiesSuperclasses;
+
+    // ORM associated sets
+    private final Set<Class<?>> convertibles;
+    private final LinkedHashSet<String> tableNames;
+    private final LinkedHashSet<String> classNames;
+
+    // Global configurations
+    private final String tablePrefix;
+
+    // State controls flow from initialization to generation
+    private boolean doneParsing;
+
+    // Current entity being parsed and found idAttributes
+    private Class<?> currentEntity;
+    private Set<Attribute> idAttributes;
+
+    @Trivial
+    public EntityParser(String tablePrefix) {
+        this.mappedSuperclasses = new HashMap<>();
+        this.entities = new HashMap<>();
+        this.embeddables = new HashMap<>();
+        this.converters = new HashMap<>();
+
+        this.entityToRecord = new HashMap<>();
+        this.entitiesSuperclasses = new ConcurrentHashMap<>();
```

Comment:
Goes along with other comment,
```suggestion
        this.entitiesSuperclasses = new HashMap<>();
```